assets/shared/components/modal/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * WordPress dependencies
+ */
+import { Icon, close as closeIcon } from '@wordpress/icons';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Modal component.
+ *
+ * @param {Object}   props
+ * @param {Function} props.onClose  Callback to run when trying to close the modal.
+ * @param {string}   props.title    The title for the modal. Empty by default.
+ * @param {Object}   props.children The content of the modal.
+ * @return {JSX.Element} The modal component or null (if not open).
+ */
+const Modal = ( { onClose, title = '', children } ) => {
+	useRunOnEscape( onClose );
+
+	return (
+		<div className={ 'sensei-modal' }>
```

Comment:
I think here we can refer to the `className` without the `{` and `}`, right?

Like this:

```suggestion
		<div className="sensei-modal">
```